runtime/nvqir/NVQIR.cpp

```diff
@@ -135,6 +136,63 @@ std::size_t qubitToSizeT(Qubit *q) {
   return q->idx;
 }
 
+template <typename T>
+concept FloatType = std::is_same<T, float>::value;
+
+template <typename T>
+concept DoubleType = std::is_same<T, double>::value;
+
+template <typename T>
+concept SimPrecisionType = FloatType<T> || DoubleType<T>;
+
+template <FloatType T>
+constexpr std::string_view typeName() {
+  return "float";
+}
+
+template <DoubleType T>
+constexpr std::string_view typeName() {
+  return "double";
+}
+
+/// Input was complex<float>/complex<double> but we prefer
+/// complex<double>/complex<float>. Make a copy, extending or truncating the
+/// values.
+template <SimPrecisionType To, SimPrecisionType From>
+std::unique_ptr<std::complex<To>[]> convertToComplex(std::complex<From> *data,
+                                                     std::size_t size) {
+  auto stateSize = pow(2, size);
+  constexpr auto toType = typeName<To>();
+  constexpr auto fromType = typeName<From>();
+  cudaq::info("copying {} complex<{}> values to complex<{}>", stateSize,
+              fromType, toType);
+
+  auto convertData = std::make_unique<std::complex<To>[]>(stateSize);
+  for (uint64_t i = 0; i < stateSize; ++i)
```

Comment:
```suggestion
  for (std::size_t_t i = 0; i < stateSize; ++i)
```
to match the declaration of stateSize.